code/game/objects/items/devices/scanners.dm

```diff
@@ -966,3 +966,35 @@ GENE SCANNER
 	playsound(src, 'sound/machines/ping.ogg', 30, TRUE)
 	return TRUE
 
+
+/obj/item/material_scanner
+	name = "material scanner"
+	desc = "A device that analyzes a object's composition and lists its materials."
+	icon = 'icons/obj/device.dmi'
+	icon_state = "forensic0"
+	item_state = "analyzer"
+	lefthand_file = 'icons/mob/inhands/equipment/tools_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/equipment/tools_righthand.dmi'
+	w_class = WEIGHT_CLASS_SMALL
+	flags_1 = CONDUCT_1
+	throwforce = 0
+	throw_speed = 3
+	throw_range = 7
+	materials = list(/datum/material/iron=30, /datum/material/glass=20)
+
+/obj/item/material_scanner/attack(obj/item/I, mob/living/user)
+	if(user.stat || user.eye_blind)
+		return
+	if(istype(I))
+		if (I.materials)
+			analyze_materials(I, user)
+			return	
+	to_chat(user, "<span class='notice'>[I] doesn't have any reusable materials.</span>")
+
+/obj/item/material_scanner/proc/analyze_materials(obj/item/I, mob/living/user)
+	to_chat(user, "- Material Analysis of [I] -")
+	for(var/R in I.materials)
+		var/datum/material/M = R
+		var/amount =  I.materials[R]
+		to_chat(user, "* [M.name] ([amount])")
+	to_chat(user, "- Analysis complete! -")
```

Comment:
I'd recommend using some spans so it's not just plain white text, maybe <span class='notice'>Or something along them lines</span>

Reply:
yea that's a good idea.

Reply:
can you make switch cases in bycode?

Reply:
Switches work in byond